unit-tests/src/test/java/org/eclipse/collections/impl/list/mutable/AbstractListTestCase.java

```diff
@@ -380,6 +380,17 @@ public void distinctWithHashingStrategy()
         Verify.assertListsEqual(FastList.newListWith("a", "b", "C", "D", "E"), actual.toList());
     }
 
+    /**
+     * @since 9.0.
+     */
+    @Test
+    public void distinctBy()
+    {
+        ListIterable<String> list = this.newWith("a", "A", "b", "C", "b", "D", "E", "e");
+        ListIterable<String> actual = list.distinctBy(String::toLowerCase);
+        Verify.assertListsEqual(FastList.newListWith("a", "b", "C", "D", "E"), actual.toList());
```

Comment:
If you use Assert.assertEquals() no need for call to toList(). It is not about efficiency, but if users refer to our tests and see something like this, it will inefficient usage.